clients/filesystem-hadoop3/src/test/java/com/datastrato/gravitino/filesystem/hadoop3/TestGravitinoVirtualFileSystem.java

```diff
@@ -187,14 +187,14 @@ public void testInternalCache() throws IOException, InterruptedException {
           localPath2.toString());
       FileSystemTestUtils.mkdirs(filesetPath2, fs);
 
-      Thread.sleep(1000);
+      Thread.sleep(5000);
       assertNull(
           ((GravitinoVirtualFileSystem) fs)
               .getFilesetCache()
               .getIfPresent(NameIdentifier.of(metalakeName, catalogName, schemaName, "fileset1")));
 
       // expired by time
-      Thread.sleep(1000);
+      Thread.sleep(5000);
```

Comment:
I think you can improve the code to use the await mechanism to check the status in a fine-grained way like below, please refer to http://www.awaitility.org/

```java
Awaitility
    .await()
    .pollDelay(4, TimeUnit.SECONDS)
    .untilAsserted(() -> Assert.assertTrue(true));
```

Reply:
I tested locally and found that using `atMost()` is more appropriate. It will continuously poll the results, thus saving some time, while `pollDelay` will force waiting for the specified time.